src/pages/iou/request/step/IOURequestStepConfirmation.tsx

```diff
@@ -177,9 +170,20 @@ function IOURequestStepConfirmation({
         }
         return reportReal ?? reportDraft;
     }, [isUnreported, shouldUseTransactionReport, transactionReport, reportReal, reportDraft]);
-    const policy = policyReal ?? policyDraft;
-    const policyID = isUnreported ? policy?.id : getIOURequestPolicyID(transaction, report);
+
+    const policy = isCreatingTrackExpense || isUnreported ? policyForMovingExpenses : (policyReal ?? policyDraft);
+    const policyID = isCreatingTrackExpense || isUnreported ? policyForMovingExpensesID : getIOURequestPolicyID(transaction, report);
     const isDraftPolicy = policy === policyDraft;
+
+    const [policyCategoriesReal] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${realPolicyID}`, {canBeMissing: true});
```

Comment:
Wouldn't it make sense to use the `policyID` here?
same for other data below.

Reply:
@parasharrajat I also changed the realPolicyID to be policyForMovingExpensesID `isCreatingTrackExpense || isUnreported ? policyForMovingExpensesID`. Take a look at the changes. Before them we also had policyID and realPolicyID. Why? I'm not sure and I guess it would take at least half a day to understand that. 😅 

For now I just want to make it work not breaking anything and causing regressions 😄 When we have the feature complete and working I would love to work on some cleanup.